src/elements/content-sidebar/MetadataSidebarRedesign.tsx

```diff
@@ -3,33 +3,150 @@
+    isBoxAiSuggestionsFeatureEnabled,
+    isFeatureEnabled,
+    onError,
+    selectedTemplateKey,
+    templateFilters,
+}) => {
+    const [editors, setEditors] = React.useState<Array<MetadataEditor>>([]);
+    const [error, setError] = React.useState<MessageDescriptor>(undefined);
+    const [file, setFile] = React.useState<BoxItem>(undefined);
+    const [templates, setTemplates] = React.useState<Array<MetadataTemplate>>([]);
+
+    const onApiError = (err: ElementsXhrError, code: string) => {
+        const { status } = err;
+        const isValidError = isUserCorrectableError(status);
+        onError(error, code, { error, [IS_ERROR_DISPLAYED]: isValidError });
+    };
+    const fetchMetadataErrorCallback = (e: ElementsXhrError, code: string) => {
+        onApiError(e, code);
+        setEditors(null);
+        setError(messages.sidebarMetadataFetchingErrorContent);
+        setTemplates(null);
+    };
+
+    const fetchMetadataSuccessCallback = ({ editors: fetchedEditors, templates: fetchedTemplates }: Metadata) => {
+        setEditors(fetchedEditors);
+        setError(undefined);
+        setTemplates(normalizeTemplates(fetchedTemplates, selectedTemplateKey, templateFilters));
+    };
+
+    const fetchMetadata = (fetchedFile: BoxItem) => {
+        if (!fetchedFile) return;
+
+        api.getMetadataAPI(false).getMetadata(
+            fetchedFile,
+            fetchMetadataSuccessCallback,
+            fetchMetadataErrorCallback,
+            isFeatureEnabled,
+            { refreshCache: true },
+        );
+    };
+
+    const fetchFileErrorCallback = (e: ElementsXhrError, code: string) => {
+        onApiError(e, code);
+        setError(messages.sidebarFileFetchingErrorContent);
+        setFile(null);
+    };
+
+    const fetchFileSuccessCallback = (fetchedFile: BoxItem) => {
+        const currentCanUpload = getProp(file, FIELD_PERMISSIONS_CAN_UPLOAD, false);
+        const newCanUpload = getProp(fetchedFile, FIELD_PERMISSIONS_CAN_UPLOAD, false);
+        const shouldFetchMetadata = !file || currentCanUpload !== newCanUpload;
+        setFile(fetchedFile);
+        if (shouldFetchMetadata) fetchMetadata(fetchedFile);
+    };
+
+    const fetchFile = () => {
+        api.getFileAPI().getFile(fileId, fetchFileSuccessCallback, fetchFileErrorCallback, {
+            fields: [FIELD_IS_EXTERNALLY_OWNED, FIELD_PERMISSIONS],
+            refreshCache: true,
+        });
+    };
+
+    React.useEffect(() => {
+        fetchFile();
+    }, []);
+
+    const showEditor = file && templates && editors;
+    const showEmptyState = showEditor && editors && editors.length === 0;
+
     return (
-        <div className="bcs-MetadataSidebarRedesign">
+        <div className="bcs-MetadataSidebarRedesign bcs-content">
             <h3>
                 <FormattedMessage {...messages.sidebarMetadataTitle} />
             </h3>
             <hr />
-            <p>Hello from Metadata Sidebar redesign</p>
+            {showEmptyState && (
```

Comment:
What do we show if showEmptyState resolves to false for some reason? Is that even possible if it gets to this point?

Reply:
we'll implement the other components in the next tickets and change this condition to if else